### PR TITLE
UI: Add `viewMode` parameter to control story nav UI

### PR DESCRIPTION
**Pull request by atanasster**
Issue: #8630

## What I did

* Handling of `viewMode` parameter in the  sidebar links
* Added parameters to the MDX stories (until now MDX were not receiving parameters in the story store)

## How to test

The subcomponents.story (Addons/Docs/ButtonGroup) will default its anchor to the `docs` tab

### CODE REVIEW

File: lib/ui/src/components/sidebar/SidebarStories.tsx [hunk 1]
```diff
@@ -52,7 +52,15 @@ const PlainLink = styled.a(plain);
 
 const Wrapper = styled.div({});
 
-const refinedViewMode = (viewMode: string | undefined, isDocsOnly: boolean) => {
+const refinedViewMode = (
+  viewMode: string | undefined,
+  isDocsOnly: boolean,
+  parameters: { viewMode?: string } = {}
+) => {
+  const { viewMode: pViewMode } = parameters;
```
Comment (shilman) [inline lib/ui/src/components/sidebar/SidebarStories.tsx]: `paramViewMode`?

Comment (atanasster) [reply lib/ui/src/components/sidebar/SidebarStories.tsx]: changed it

**Comment by shilman** [comment]
@atanasster thanks for figuring this out. i cleaned up the code slightly and added tests/documentation.

**Comment by atanasster** [comment]
Thanks, one down.... :)

**Comment by LarsDenBakker** [comment]
This is great! Thanks!

**Comment by wagerfield** [comment]
This is not working for me. I've tried:

1. Setting `viewMode` in the preview/manager via `addParameters`

```js
// .storybook/preview.js
import { addParameters } from "@storybook/react"

addParameters({ viewMode: "docs" })
```

2. In a story file's CSF meta export:

```js
// src/componenets/component.stories.js

export default {
  parameters: {
    viewMode: "docs"
  }
}
```

3. On a story's meta/config object:

```js
// src/componenets/component.stories.js

const ComponentStory = () => <Component />

ComponentStory.story = {
  parameters: {
    viewMode: "docs"
  }
}
```

...none of these variations are working for me? 

I'm using the latest version (5.3.14) of Storybook and the official addons inc. docs.

**Comment by shilman** [comment]
@wagerfield this is a 6.0 feature

**Comment by wagerfield** [comment]
Thanks @shilman, I've upgraded to `@next` and it's working like a charm :D

**Comment by marchaos** [comment]
I can't get this to work on the latest alpha (31). Checking here https://next--storybookjs.netlify.com/official-storybook/?path=/story/addons-docs-buttongroup--basic and it does not seem to work either. Can you confirm this?

**Pull request merged**